mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractAttachmentMapperTest.java

```diff
@@ -60,46 +65,113 @@ public void tearDown() throws MailboxException {
     }
 
     @Test
-    public void addShouldWork() throws IOException {
+    public void putShouldWork() throws IOException {
         // Given
-        String blobId = "blobId";
-        String expected = "content";
+        InMemoryAttachmentId blobId = InMemoryAttachmentId.of(1);
+        String expectedContent = "content";
+        Content content = new ByteContent(expectedContent.getBytes(Charsets.UTF_8));
```

Comment:
I just read the code, and you should not use Byte content ...

We might want to be able to Stream content in a near future (Cassandra best practice for instance). ByteContent is backed by a byte array.

I would rather see something that takes an input stream and a size (so that we do not need to keep the whole content in memory). It would allow implementations to be more efficient. 

2 contructors : 
 - one with size and InputSream
 - one with InputStream (use TeeInputStream for size calculation )